core/src/main/scala/com/microsoft/azure/synapse/ml/explainers/ICEExplainer.scala

```diff
@@ -41,6 +44,11 @@ trait ICEFeatureParams extends Params with HasNumSamples {
     this.setCategoricalFeatures(features)
   }
 
+  def setCategoricalFeaturesR(jsonString: String): this.type = {
```

Comment:
likewise here if possible remove these APIs and call json parse from R side in the generated code